core/templates/dev/head/pages/email-dashboard-pages/email-dashboard-data.service.ts

```diff
@@ -16,109 +16,123 @@
+      params: params
+    }).toPromise();
+  }
 
-    return {
-      getQueries: function() {
-        return queries;
-      },
-
-      getCurrentPageIndex: function() {
-        return currentPageIndex;
-      },
-
-      getLatestCursor: function() {
-        return latestCursor;
-      },
-
-      submitQuery: function(data) {
-        var startQueryIndex = currentPageIndex * QUERIES_PER_PAGE;
-        var endQueryIndex = (currentPageIndex + 1) * QUERIES_PER_PAGE;
-
-        return $http.post(QUERY_DATA_URL, {
-          data: data
-        }).then(function(response) {
-          var data = response.data;
-          var newQueries = [data.query];
-          queries = newQueries.concat(queries);
-          return queries.slice(startQueryIndex, endQueryIndex);
-        });
-      },
-
-      getNextQueries: function() {
-        var startQueryIndex = (currentPageIndex + 1) * QUERIES_PER_PAGE;
-        var endQueryIndex = (currentPageIndex + 2) * QUERIES_PER_PAGE;
-
-        if (queries.length >= endQueryIndex ||
-            (latestCursor === null && currentPageIndex !== -1)) {
-          currentPageIndex = currentPageIndex + 1;
-          return $q(function(resolver) {
-            resolver(queries.slice(startQueryIndex, endQueryIndex));
-          });
-        } else {
-          currentPageIndex = currentPageIndex + 1;
-          return fetchQueriesPage(QUERIES_PER_PAGE, latestCursor)
-            .then(function(data) {
-              queries = queries.concat(data.recent_queries);
-              latestCursor = data.cursor;
-              return queries.slice(startQueryIndex, endQueryIndex);
-            });
-        }
-      },
-
-      getPreviousQueries: function() {
-        var startQueryIndex = (currentPageIndex - 1) * QUERIES_PER_PAGE;
-        var endQueryIndex = currentPageIndex * QUERIES_PER_PAGE;
-        currentPageIndex = currentPageIndex - 1;
-        return queries.slice(startQueryIndex, endQueryIndex);
-      },
-
-      isNextPageAvailable: function() {
-        var nextQueryIndex = (currentPageIndex + 1) * QUERIES_PER_PAGE;
-        return (queries.length > nextQueryIndex) || Boolean(latestCursor);
-      },
-
-      isPreviousPageAvailable: function() {
-        return (currentPageIndex > 0);
-      },
-
-      fetchQuery: function(queryId) {
-        return $http.get(QUERY_STATUS_CHECK_URL, {
-          params: {
-            query_id: queryId
-          }
-        }).then(function(response) {
-          var data = response.data;
-          queries.forEach(function(query, index, queries) {
-            if (query.id === queryId) {
-              queries[index] = data.query;
-            }
-          });
-          return data.query;
+  getQueries(): Array<Query> {
+    return this.queries;
+  }
+
+  getCurrentPageIndex(): number {
+    return this.currentPageIndex;
+  }
+
+  getLatestCursor(): string {
+    return this.latestCursor;
+  }
+
+  submitQuery(data: any): Promise<Array<Query>> {
```

Comment:
Is this any really? If not please add a TODO

Reply:
Done